### Mutation {#seq_determ_mut}

In the previous section, we looked at the competition between two organisms
carrying different alleles, $A$ and $a$. But we completely ignored how these two
versions of a particular genetic element appeared in the population. Recall from
[`cite chapter on ingredients of evolution`] that one of the ingredients for
evolution is the generation of variability between organisms. We now know that
this variability comes from modifications in the genetic sequence----mostly DNA,
but RNA for some viruses---that happens randomly as they replicate their
genomes, as we saw in [`cite Luria-Delbruck section`]. In this section, we will
mathematize mutation as the second evolutionary force shaping the structure of
populations. But why do we include mutation as a deterministic evolutionary
force if mutations happen randomly? As we mentioned in @sec-selection, we
imagine studying a very large microbial population whose size we take to be a
continuous variable. In this regime, random mutations in individual organisms
are blurred as we quantify them as a deterministic fraction of the population
switching alleles.

![One-locus two-allele mutation. Schematic of the toy model for mutation only
case.](./figures/placeholder.jpg){#fig-mutation_schematic}

For our one-locus two-allele toy model, we think of mutations between alleles
$A$ and $a$ as a simple first-order chemical reaction of the form
$$
A
\underset{\mu_{A\rightarrow a}}{\stackrel{\mu_{a\rightarrow A}}{\rightleftharpoons}}
a,
$${#eq-mut_rxn}
where $\mu_{A\rightarrow a}$ is the rate at which organisms carrying allele $A$
mutate to allele $a$ and $\mu_{a\rightarrow A}$ is the equivalent mutation rate
from $a$ to $A$. It is useful to interpret the mutation rate as the probability
of switching allele per unit of time. That means that on a small time window
$\Delta t$ the probability of a single organism switching from allele $A$ to $a$
is given by
$$
P_{\Delta t}(A \rightarrow a) = \mu_{A\rightarrow a} \Delta t.
$${#eq-prob_mutation}
From this equation, we can see that the mutation rate has units
$\mu_{A\rightarrow a} [=] \frac{\text{\#organisms}}{\text{time}}$. To compute
the number of organisms with allele $A$ at time $t + \Delta t$, we take into
account both the number of organisms switching from $a$ to $A$ and from $A$ to
$a$; this is
$$
N_A(t + \Delta t) = N_A(t) + 
\overbrace{N_a(t)\mu_{a\rightarrow A} \Delta t}^{\text{gain } a \rightarrow A}
-\overbrace{N_A(t)\mu_{A\rightarrow a}\Delta t}^{\text{loss } A \rightarrow a}.
$${#eq-NA_t_delta_t_mut}
Just as in @eq-NA_t_delta_t, we can send the first term on the right-hand side
of @eq-NA_t_delta_t_mut to the left, and divide both sides by $\Delta t$. For
very small $\Delta t$, we can rewrite @eq-NA_t_delta_t_mut as a differential
equation of the form
$$
\frac{dN_A}{dt} = N_a(t)\mu_{a\rightarrow A} - N_A(t)\mu_{A\rightarrow a}.
$${#eq-dNA_dt_mut}
We can write an equivalent equation for $N_a$ where the signs would be simply
flipped since every time an organism is lost from allele $A$ it must be gained
in allele $a$, and vice versa. Since the equation for $\frac{dN_a}{dt}$ is 
given by $-\frac{dN_A}{dt}$, for this mutation-only scenario, the total number
of organisms $N_\text{tot}$ does not change with time.

Let us now write the dynamics for the allele frequency $x(t)$. As before, we
simply substitute the definition of $x(t)$ on the differential equation, 
obtaining
$$
\frac{dx}{dt} = \frac{d}{dt} \left( \frac{N_A}{N_\text{tot}} \right) = 
\frac{1}{N_\text{tot}} \frac{dN_A}{dt},
$${#eq-dx_dt_def_mut}
where we took $N_\text{tot}$ out of the derivative since the total number of
organisms doesn't change over time. Substituting @eq-dNA_dt_mut into
@eq-dx_dt_def_mut results in
$$
\frac{dx}{dt} = - \mu_{A\rightarrow a} x + \mu_{a\rightarrow A} (1 - x),
$${#eq-mutation_ode}
where we again suppressed the time dependence for notation simplicity. Before
attempting to solve @eq-mutation_ode let us take a look at the steady-state 
value for $x(t)$. This means that we imagine this process of mutating back and
forth between both alleles happening for a long time until it reaches a point
where it does not change anymore. To compute the steady-state, we set the time
derivative to zero, obtaining
$$
\mu_{A\rightarrow a} x_{ss} = \mu_{a\rightarrow A} (1 - x_{ss}),
$${#eq-steady_state_mut}
where the subscript $ss$ reminds us this is the steady state allele frequency.
If we now solve for $x_{ss}$ this results in
$$
x_{ss} = \frac{\mu_{a\rightarrow A}}{\mu_{a\rightarrow A}+\mu_{A\rightarrow a}}.
$${#eq-x_steady_state_mut}
Thus, the steady-state allele frequency is given by the ratio of the mutation
rate from $a$ to $A$ to the sum of the back and forth mutation rates. This
result implies that in the limit where it is much more likely to mutate from $a$
to $A$, i.e., $\mu_{a\rightarrow A} \gg \mu_{A\rightarrow a}$ the allele
frequency goes to one, i.e., allele $A$ will go into fixation in the population.
The opposite would be true for a much larger mutation rate from $A$ to $a$.

Now let's go ahead and solve the dynamics from @eq-mutation_ode. We can solve
this equation using the separation of variables method. We separate terms with
the allele frequency to the left and the time-dependent ones to the right, and
integrate both sides, obtaining
$$
\int \frac{dx}{- \mu_{A\rightarrow a} x + \mu_{a\rightarrow A} (1 - x)} = 
\int dt.
$${#eq-dx_dt_sep_variables_mut}
To integrate the left-hand side we define
$$
u \equiv - \mu_{A\rightarrow a} x + \mu_{a\rightarrow A} (1 - x),
$${#eq-dx_dt_sep_variables_mut_2}
and compute $du$ as
$$
du = \left( - \mu_{A\rightarrow a} - \mu_{a\rightarrow A} \right) dx.
$${#eq-dx_dt_sep_variables_mut_3}
We can then substitute @eq-dx_dt_sep_variables_mut_2 and
@eq-dx_dt_sep_variables_mut_3 into @eq-dx_dt_sep_variables_mut, obtaining
$$
-\frac{1}{\mu_{a\rightarrow A} + \mu_{A\rightarrow a}} \int \frac{du}{u} = 
\int dx.
$${#eq-dx_dt_sep_variables_mut_4}
From @eq-dx_dt_sep_variables_mut_4 is easy to see that the integral in
@eq-dx_dt_sep_variables_mut results in
$$
-{1 \over \mu_{a\rightarrow A} + \mu_{A\rightarrow a}}
\ln \left( \mu_{A a} x - \mu_{a A} (1 - x) \right) =
t + C,
$${#eq-mutation_int}
where $C$ is an integration constant. Substituting the initial condition $x(t=0)
= x_o$ we have that this constant takes the form
$$
C =  -{1 \over \mu_{a\rightarrow A} + \mu_{A\rightarrow a}}
\ln \left( \mu_{A\rightarrow a} x_o - \mu_{a\rightarrow A} (1 - x_o) \right).
$${#eq-mutation_const}
Substituting @eq-mutation_const into @eq-mutation_int results in
$$
-\frac{1}{\mu_{a\rightarrow A} + \mu_{A\rightarrow a}} \ln \left( 
\frac{ \mu_{A\rightarrow a} x - \mu_{a\rightarrow A} (1 - x)}
{\mu_{A\rightarrow a} x_o - \mu_{a\rightarrow A} (1 - x_o)} 
\right) = t.
$${#eq-mutation_sol}
Exponentiating both sides of @eq-mutation_sol gives 
$$
\left( \frac{ \mu_{A\rightarrow a} x - \mu_{a\rightarrow A} (1 - x)}
{\mu_{A\rightarrow a} x_o - \mu_{a\rightarrow A} (1 - x_o)} \right)^{
-{1 \over \mu_{a\rightarrow A} + \mu_{A\rightarrow a}}} = \mathrm{e}^t.
$${#eq-mutation_sol_2}
Raising both sides of @eq-mutation_sol_2 to the $\mu_{a A} + \mu_{A\rightarrow
a}$ power resulst in
$$
\left( \frac{ \mu_{A\rightarrow a} x - \mu_{a\rightarrow A} (1 - x)}
{\mu_{A\rightarrow a} x_o - \mu_{a\rightarrow A} (1 - x_o)} \right) =
\mathrm{e}^{- (\mu_{a A} + \mu_{A\rightarrow a}) t}.
$${#eq-mutation_sol_3} 
We can now solve for $x(t)$, and after a little messy algebra, we find that 
$$
x(t) = \frac{\mu_{a\rightarrow A}}
{\mu_{a\rightarrow A} + \mu_{A\rightarrow a}} 
- \frac{\left[  \mu_{a\rightarrow A} - (\mu_{a\rightarrow A} + 
\mu_{A\rightarrow a}) x_o \right] 
\mathrm{e}^{- (\mu_{a\rightarrow A} + \mu_{A\rightarrow a}) t}}
{\mu_{a\rightarrow A} + \mu_{A\rightarrow a}}.
$${#eq-allele_freq_mut}
One can argue that @eq-allele_freq_mut does not give us a clear intuition of
what the dynamics of the allele frequency will behave like. But we can easily
see the second term on the right-hand side goes to zero in the limit when $t
\rightarrow \infty$, leaving behind the steady state solution we found in
@eq-x_steady_state_mut. @fig-mutation_dynamics shows the dynamics for different
ratios of the mutation rates. Notice that, as expected, the allele frequency
converges to the steady state value we derived.

![Allele frequency dynamics under mutation only. Each curve shows the time
evolution of allele $A$ frequency for different ratios of mutation rates. In the
long term, all frequencies reach the steady sate predicted by
@eq-x_steady_state_mut.](./figures/forces_evolution/_mutation_dynamics){#fig-mutation_dynamics}

### Combining evolutionary forces: mutation-selection balance

So far, in this section and in @sec-selection, we modeled two of the
evolutionary forces as deterministically modifying the population structure.
Given that both selection and mutation are modeled in the same framework, we can
combine them into a single description of the dynamics to study the effects of
having both forces acting on a population. As we will see in later chapters,
this regime is relevant for viruses with large populations and strong selection
from the immune system acting on them. The dynamics for our single-locus
two-allele system, subject to selection and mutation, is given by the sum of the
right-hand side of @eq-dNA_dt_mut and @eq-mutation_ode. This is
$$
\frac{dx}{dt} = s x (1 - x) - 
\mu_{A\rightarrow a} x + \mu_{a\rightarrow A} (1 - x).
$${#eq-sel_mut} 
This linear ODE can be solved analytically as you can assess yourself by plugin
@eq-sel_mut into a symbolic math package such as `Mathematica`. The problem is
that the solution is not intuitive, with a bunch of tangents and arctangents
that make it very difficult to look at. Nevertheless, we can still gain
intuition about the long-term behavior of @eq-sel_mut. An interesting setting to
analyze is the case where one of the alleles, let's say $a$ is deleterious. For
this case, our results from @sec-selection tell us that natural selection would
remove this allele from the population, leading $A$ to fixation in the
population. But if mutation keeps bringing the allele back repeatedly, a balance
must exist between mutation and selection. This is a plausible scenario if we
think of allele $A$ as the functional version of a particular protein, and $a$
as a coarse-grained state of all non-functional proteins, for example. To
compute this equilibrium, we need to compute @eq-sel_mut's steady-state by
setting the left-hand side to zero
$$
0 = s x_{ss}^2 + (\mu_{Aa} + \mu_{aA} - s) x_{ss} - \mu_{aA},
$${#eq-sel_mut_steady_state} 
where we already expanded the terms and group by powers of $x_ss$, the steady
state allele frequency. To solve for $x_{ss}$, we compute the roots of this
quadratic equation, obtaining
$$
x_{ss} = \frac{(s - \mu_{A\rightarrow a} - \mu_{a\rightarrow A}) \pm 
\sqrt{(\mu_{A\rightarrow a} - \mu_{a\rightarrow A} - s)^2 + 
4 s \mu_{a\rightarrow A}}}{2s}.
$${#eq-sel_mut_steady_state_roots} 
Given that we take allele $a$ to be a coarse-grained state for all
non-functional versions of the gene, which we expect to exceed in number the
functional versions by several orders of magnitude [`Look for experimental
evidence of this`], we can safely assume that $\mu_{a\rightarrow A} \approx 0$.
In other words, there are many more ways to break the protein than to bring it
back to its functional form. This assumption results in
$$
x_{ss} = \frac{(s - \mu_{A\rightarrow a}) \mp (s - \mu_{A\rightarrow a})}{2s}.
$${#eq-sel_mut_roots_approx}
The resulting two possible steady states are then
$$
x_{ss}^{(1)} = 0, \quad x_{ss}^{(2)} = 1 - \frac{\mu_{A\rightarrow a}}{s}.
$${#eq-sel_mut_roots} 
The first case in @eq-sel_mut_roots shows that if allele $A$ were to go extinct,
it would stay as such, not being able to come back into the population. However,
the more interesting and relevant case is the second root. This shows that the
beneficial allele $A$ frequency decreases from the fixation value $x_{ss} = 1$
depending on the relative strengths of mutation and selection.
