### Natural selection (survival of the fittest) {#sec-selection}

1859 marked the year that Charles Darwin shocked the world with the incredible
insights in his book *On the origin of Species*. The complete title of the book
*On the Origin of Species by Means of Natural Selection, or the Preservation of
Favoured Races in the Struggle for Life* perfectly captures Darwin's proposed
mechanism for how evolution takes place. Nowadays, with the so-called
modern-synthesis of evolution, we know natural selection is not the only
evolutionary force that shapes living organisms. In the coming sections, we will
mathematically explore mechanisms such as mutation and genetic drift that all
contribute to forming the many *endless forms, most beautiful* that live on our
planet. But we will begin with the first evolutionary force discovered by Darwin
and Alfred Russel Wallace.

Natural selection is intrinsically associated with the concept of fitness. The
phrase "*survival of the fittest*," first used by Darwin, guided and still
guides the way that biologists think about the evolution of many organisms. But
even though fitness is part of the daily jargon of many biologists, it is a
subtle and highly debated concept. After all, what defines the ability of an
organism to survive the challenges surrounding them are entirely
context-dependent. Roughly speaking, we can think of fitness as the ability of
an organism or a population of organisms to survive and reproduce in the given
ecological niche they occupy. The term ecology must be included because
fitness results from the interplay between organisms and their environment,
including all biotic and abiotic interactions.

It is common both in theory and in experiments to use the relative growth rates
of organisms, i.e., the speed at which they can reproduce and generate
offspring, as a proxy for fitness. This is a convenient approximation, but we
should not lose track of the relevant context-dependence on what fitness can
mean. Just because redwoods have an average life span of 500-700 years and a
meager growth rate, that does not mean they are not fit. Having said that, we
will begin with the simplest form of fitness, i.e., frequency-independent
selection. The term frequency-independence refers to the assumption that the
fitness of a particular allele does not depend on its own relative abundance.
This assumption could break down for cases such as some pathogenic bacteria
coordinating their attack via cell-to-cell communication known as quorum
sensing. But we need to learn how to crawl before we try to run.

#### Frequency independent selection (haploids) {#sec-freq_indep_haplo}

Because of our love for bacteria and the fact that many of the most exciting
experiments in evolution have been and are being done in haploid organisms, we
first focus our efforts on these single-cell organisms. Haploidy refers to the
biological feature of having a single copy of a chromosome. As bacteria divide
into two cells, they segregate (split) one copy of the genome into each daughter
cell (extrachromosomal elements such as plasmids behave differently). This
contrasts with diploidy, where organisms such as humans receive one copy of each
chromosome from each parent. What haploidy means for our task of modeling
selection on a one-locus two-allele system is that we only need to focus on the
fitness value of each allele without having to worry about a combination of
alleles.

For the simplest case of one-locus two-alleles, let us label the two alleles as
$A$ and $a$. Let $N_A(t)$ and $N_a(t)$ be the corresponding number of organisms
carrying each allele at time $t$. We want to compute the number of organisms
with, let's say, allele $A$ after a time $\Delta t$. For the
frequency-independent selection, the only factor that matters is the current
number of organisms with allele $N_A(t)$ and their corresponding growth rate
$f_A$. As discussed before, we will use this growth rate as a proxy for fitness;
this is sometimes called Malthusian fitness after Thomas Robert Malthus---famous
scholar whose essay on population growth inspired both Darwin and Wallace. This
growth rate has units of $f_A [=] \frac{\#\text{organisms}}{\text{time}}$.
Therefore, the number of organisms at time $t + \Delta t$ can then be computed
as
$$
N_A(t + \Delta t) = \overbrace{N_A(t)}^\text{current number of organisms}
+ \overbrace{f_A N_A(t) \Delta t}^{\text{change in number in }\Delta t}.
$${#eq-NA_t_delta_t}
Sending the first term in @eq-NA_t_delta_t to the right and assuming $\Delta t$
is a tiny time window (formally assuming the limit $\Delta t \rightarrow 0$)
allows us to write an ordinary differential equation of the form
$$
\frac{d N_A}{dt} = f_A N_A(t).
$${#eq-expo_growth}
We can define an equivalent equation for allele $a$ with fitness $f_a$.

Rather than tracking the *absolute number* of organisms carrying a particular
allele, it is useful to track the relative abundances of each allele. This is
especially important when dealing with diploid organisms, which have two copies
of each gene. Therefore, we define $N_{\text{tot}}(t) \equiv N_A(t) + N_a(t)$ as
the total number of organisms and the relative abundance of allele $A$, $x(t)$,
as
$$
x(t) = \frac{N_A(t)}{N_\text{tot}(t)}.
$${#eq-allele_frac_def}
We can write an equivalent equation to @eq-expo_growth for $N_\text{tot}(t)$.
But, in this case, the change in the total number of organisms is the sum of the
changes in both types, i.e.,
$$
\frac{d N_\text{tot}}{dt} = \frac{dN_A}{dt} + \frac{dN_a}{dt} 
= f_A N_A(t) + f_a N_a(t),
$${#eq-dNtot_dt}
where for the second equality, we substituted the right-hand side of
@eq-expo_growth for each of the alleles. The time dynamics for $N_\text{tot}(t)$
are of the form
$$
\frac{dN_\text{tot}}{dt} = \bar{f}(t) N_\text{tot}(t),
$${#eq-dNtot_fbar}
where $\bar{f}(t)$ is the average population fitness at time $t$. We can deduce
the value of $\bar{f}(t)$ if we notice that the dynamics of the total population
is the sum of the individual alleles dynamics. This implies we can factorize an
$N_\text{tot}(t)$ out of the right-hand side of @eq-dNtot_dt using
@eq-allele_frac_def, obtaining
$$
\frac{dN_\text{tot}}{dt} = 
N_\text{tot}(t) \left[ f_A x(t) + f_a (1 - x(t)) \right].
$${#eq-dNot_factorized}
From this, we can see the average fitness $\bar{f}(t)$ is of the form
$$
\bar{f}(t) = f_A x(t) + f_a (1 - x(t)).
$${#eq-pop_fitness_def}

Since we care about the relative abundance of allele $A$, $x(t)$, we can take
the time derivative on both sides of @eq-allele_frac_def
$$
\frac{dx}{dt} = \frac{d}{dt}\left( {N_A(t) \over N_\text{tot}(t)} \right).
$${#eq-dx_dt}
Applying the chain rule on the right-hand side of @eq-dx_dt results in
$$
\frac{dx}{dt} = \frac{\dot{N}_A(t) N_\text{tot}(t) - 
N_A(t) \dot{N}_\text{tot}(t)}{N_\text{tot}(t)^2},
$${#eq-dx_dt_chainrule}
where we define $\dot{N}_x = dN_x / dt$ for brevity. Substituting the time
derivatives from @eq-expo_growth and @eq-allele_frac_def on @eq-dx_dt_chainrule
gives
$$
\frac{dx}{dt} = \frac{\left[f_A N_A(t)\right] 
N_\text{tot}(t) - N_A(t)[\bar{f}(t) N_\text{tot}(t)]}{N_\text{tot}(t)^2}.
$${#eq-dx_dt_chainrule_2}
We now substitute the definition of the average fitness from @eq-pop_fitness_def
to obtain
$$
\frac{dx}{dt} = \frac{f_A N_A(t) - N_A(t) [f_a x(t) + f_a (1 - x(t))]}
{N_\text{tot}(t)}.
$${#eq-dx_Dt_chainrule_3}
This can be further simplified by using the definition of the allele frequency
from @eq-allele_frac_def to obtain
$$
\frac{dx}{dt} = f_A x(t) - x(t) \left[f_A x(t) + f_a (1 - x(t))\right].
$${#eq-dx_Dt_chainrule_4}
Grouping terms by the fitness values results in
$$
\frac{dx}{dt} = f_A x(t)(1 - x(t)) - f_a x(t)(1 - x(t)) = s x(t)(1 - x(t)),
$${#eq-freq_dynamics_selection}
where we defined $s = f_A - f_a$, the fitness difference between alleles, also
known as the *selection coefficient*. @eq-freq_dynamics_selection is telling us
that for the population to change its relative composition---what we define as
the evolution of the population---there must be a difference in fitness (growth
rates) between the competing organisms. 

We can solve @eq-freq_dynamics_selection equation for the specific case where
$s$ is a constant, meaning that the fitness values of the alleles don't change
over time. Specifically, we can use the separation of variables, and integrate
on both side, obtaining
$$
\int \frac{dx}{x(1 - x)} = s\int dt,
$${#eq-sep_var}
where we drop the time dependence for notation convenience. For the left-hand
side of @eq-sep_var, let us use partial fractions to write the integrand as a
sum of two terms of the form
$$
\frac{1}{x(1 - x)} = \frac{A}{x} + \frac{B}{(1 - x)}
= \frac{A - Ax + Bx}{x(1 - x)},
$${#eq-partial_frac}
where $A$ and $B$ are coefficients to be found. Substituting $A = 1$ and $B = 1$
in @eq-partial_frac we have
$$
\frac{1 - x + x}{x(1 - x)} = \frac{1}{x(1 - x)},
$${#eq-partial_frac_2}
thus, these coefficients recover the left-hand side of @eq-partial_frac. This
allows us to split @eq-sep_var into two integrals of the form
$$
\int \frac{dx}{x} + \int \frac{dx}{1 - x} = s \int dt.
$${#eq-sep_var_split}
Integrating both sides of @eq-sep_var_split results in
$$
\ln x - \ln(1 - x) = \ln\left( {x \over 1 - x} \right) = s t + C,
$${#eq-sep_var_sol}
where $C$ is an integration constant for which we need initial conditions. If we
set an initial condition $x(t = 0) = x_o$ we find that
$$
C = \ln \left( \frac{x_o}{1 - x_o} \right).
$${#eq-const_int_fitness}
Substituting @eq-const_int_fitness in @eq-sep_var_sol and exponentiating both
sides gives
$$
\frac{x}{1 - x} = \frac{x_o}{1 - x_o} \mathrm{e}^{s t}.
$${#eq-sep_var_int}
We can now solve @eq-sep_var_int for $x$. First, we send the denominator on the
left-hand side to the right, obtaining
$$
x = (1 - x)\left( x_o \over 1 - x_o \right)\mathrm{e}^{st}
$${#eq-fitness_solve_x}
Next, we expand the product on the right-hand side of @eq-fitness_solve_x and
send the terms with $x$ back to the left, resulting in
$$
x + x \frac{x_o}{1 - x_o}\mathrm{e}^{st} = \frac{x_o}{1 - x_o}\mathrm{e}^{st}.
$${#eq-fitness_solve_x_2}
Factorizing $x$ in @eq-fitness_solve_x_2 results in
$$
x \left( 1 + \frac{x_o}{1 - x_o}\mathrm{e}^{st} \right) = 
\frac{x_o}{1 - x_o}\mathrm{e}^{st}.
$${#eq-fitness_solve_x_3}
The term in parenthesis on the left-hand side of @eq-fitness_solve_x_3 can be
rewritten as
$$
x \left( \frac{1 - x_o + x_o\mathrm{e}^{st}}{1 - x_o} \right) =
\frac{x_o}{1 - x_o}\mathrm{e}^{st}.
$${#eq-fitness_solve_x_4} 
This allows us to cancel the denominator on both sides of the equation to obtain
$$
x \left[ 1 + x_o \left( \mathrm{e}^{st} - 1 \right) \right] = 
x_o \mathrm{e}^{st}.
$${#eq-fitness_solve_x_5}
Finally, we can solve @eq-fitness_solve_x_5 for the allele frequency $x(t)$,
obtaining our final 
result
$$
x(t) = \frac{x_o \mathrm{e}^{st}}{1 + x_o \left( \mathrm{e}^{st} - 1 \right)}.
$${#eq-allele_freq_result}
@eq-allele_freq_result is the solution of the so-called logistic equation.
@fig-logistic_allele shows this solution for different selection coefficients.
In this deterministic regime, the population's fate is set as long as there is a
fitness difference, i.e., as long as $s \neq 0$. If $s > 0$ allele $A$ will go
into fixation. Otherwise, if $s < 0$, the allele will go extinct.

![**One-locus two-alleles model allele frequency dynamics**. Change in allele
frequency over time as defined by @eq-allele_freq_result for different selection
coefficients.](./figures/forces_evolution/_selection_logistic){#fig-logistic_allele}